src/main/kotlin/com/terraformation/backend/customer/model/IndividualUser.kt

```diff
@@ -365,9 +365,15 @@ data class IndividualUser(
   }
 
   override fun canUpdateAccession(accessionId: AccessionId): Boolean {
-    // All users in a project can write all accessions in the project's facilities, so this
-    // is the same as the read permission check.
-    return canReadAccession(accessionId)
+    // Updating accession is allowed only for Role.MANAGER and higher.
+    // The user must also have read permissions for the accession.
+    val facilityId = parentStore.getFacilityId(accessionId)
+    return when (facilityRoles[facilityId]) {
+      Role.OWNER,
+      Role.ADMIN,
+      Role.MANAGER -> true
+      else -> false
+    } && canReadAccession(accessionId)
```

Comment:
We don't need the read permission check here; it'll cause an extra `parentStore` data fetch on the happy path and will never be false when the `when` expression is true. The test suite should ensure the return value is always what we want.